crypto/evp/exchange.c

```diff
@@ -202,8 +202,8 @@ int EVP_PKEY_derive_init(EVP_PKEY_CTX *ctx)
 
     /* Ensure that the key is provided.  If not, go legacy */
     tmp_keymgmt = ctx->keymgmt;
-    provkey = evp_pkey_make_provided(ctx->pkey, ctx->libctx,
-                                     &tmp_keymgmt, ctx->propquery);
+    provkey = evp_pkey_export_to_provider(ctx->pkey, ctx->libctx,
+                                          &tmp_keymgmt, ctx->propquery);
     if (provkey == NULL)
         goto legacy;
     if (!EVP_KEYMGMT_up_ref(tmp_keymgmt)) {
```

Comment:
why is this one uprefed bu tthe set_peer one isnt?

Reply:
Because the set_peer one doesn't init the ctx